cli_tools_e2e_test/gce_windows_upgrade/test_suites/windows_upgrade/windows_upgrade_test_suite.go

```diff
@@ -322,6 +322,17 @@ func runTest(ctx context.Context, image string, args []string, testType utils.CL
 		return
 	}
 
+	defer func() {
+		// delete the test instance when test is done
+		if !utils.RunTestCommandIgnoringError("gcloud", []string{
+			"compute", "instances", "delete",  "--quiet",
+			fmt.Sprintf("--zone=%v", testProjectConfig.TestZone),
+			fmt.Sprintf("--project=%v", testProjectConfig.TestProjectID), instanceName,
+		}, logger, testCase) {
+			return
+		}
```

Comment:
Is there a more elegant way to delete instances, e.g. by calling compute client directly? I found this call very confusing as it appears it's running a test, but it's actually running `gcloud compute instances delete`. At minimum, this should be extracted in an appropriately named func, but it would be nicer not to yse RunTest syntax for cleanup tasks.

Reply:
Thank you for the advice, I wrapped it with a "cleanupTestInstance" func.
My consideration here is that since it's created by gcloud successfully, it very probably can be deleted by gcloud successfully . Using a compute client requires us to init it by align credentials, which have a chance to make unnecessary mistake.

Reply:
OK. Just avoid calling RunTest if possible. RunCommand or similar is fine.

Reply:
Maybe export runCliTools from utils and use that?

Reply:
Avoided in the main flow. The func name self explained the purpose.

Reply:
done